qiskit_experiments/framework/json.py

```diff
@@ -254,22 +254,14 @@ def _deserialize_type(value: Dict):
             method_cls = None
             name = qualname[0]
         mod = value["module"]
+        mod_scope = importlib.import_module(mod)
         scope = None
-        if mod == "__main__":
-            if method_cls is None:
-                if name in globals():
-                    return globals()[name]
-            else:
-                scope = globals().get(method_cls, None)
+        if method_cls is None:
+            scope = mod_scope
         else:
-            mod_scope = importlib.import_module(mod)
-            if method_cls is None:
-                scope = mod_scope
-            else:
-                for name_, obj in inspect.getmembers(mod_scope, inspect.isclass):
-                    if name_ == method_cls:
-                        scope = obj
-
+            for name_, obj in inspect.getmembers(mod_scope, inspect.isclass):
```

Comment:
`dict(inspect.getmembers(mod_scope, inspect.isclass)).get(method_cls, None)`

which one is faster? (in ``__main__`` we often have tons of items)

Reply:
I would expect constructing dict would be worse since it has to iterate over everything first anyway and then do lookup, when the for loop can at least terminate early.

Generally I think while main might have tons of items, not a lot of them will be classes so this iteration shouldn't be too bad.

Reply:
Fair enough.